konflux: operator build triggers

littlejawa opened the pull request:
Modify the on-cel-expressions to fix triggering issues.
We need to prevent the operator from building on changes to the bundle, as otherwise we have a loop of builds/updates.

littlejawa commented:
/retest

littlejawa commented:
Example of a PR that triggers a rebuild of the operator because of the config/manager/manager.yaml change:
https://github.com/openshift/sandboxed-containers-operator/pull/654

spotlesstofu commented on .tekton/osc-operator-pull-request.yaml:
```diff
@@ -14,7 +14,7 @@ metadata:
       files.all.exists(path, !path.matches('config/peerpods/podvm/.*|.tekton/.*podvm-builder.*yaml$')) &&
       files.all.exists(path, !path.matches('.tekton/.*fbc-.*|fbc/.*')) &&
-      files.all.exists(path, !path.matches('config/manager/.*'))
+      files.all.exists(path, !path.matches('config/manager/manager.yaml'))
```
What about moving the `!` like below?

```suggestion
      !files.all.exists(path, path.matches('config/manager/.*'))
```

This should provide more predictable outputs. I tested it in [CEL Playground](https://playcel.undistro.io/).

littlejawa replied:
It's certainly worth a try :)
Now I'm wondering if I should do the same to the other filters? They *seem* to work, but would that be more reliable if I move the `!` there too?

spotlesstofu replied:
yes, I'd do the same on all of them

spotlesstofu commented:
The enterprise contract fails because it now checks against the release policy, I merged that earlier today.
We can address the violations in a separate pull request.

littlejawa commented:
> The enterprise contract fails because it now checks against the release policy, I merged that earlier today. We can address the violations in a separate pull request.

Right. I started looking into it - seems like I need to add some labels everywhere :-)

littlejawa committed:
konflux: fix build triggers for the operator

Our operator gets rebuilt when some unrelated files are modified.
This is typically the case when the bundle is updated, which is a problem
as if the operator gets rebuilt when the bundle in updated, this triggers
a loop of build/updates.

This commit modifies the way we check for the files that should be ignored
when deciding when to build. Hopefully, this will fix the problem we are seeing.

Signed-off-by: Julien Ropé <jrope@redhat.com>
